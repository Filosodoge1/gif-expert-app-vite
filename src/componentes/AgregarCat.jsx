import React, { useState } from 'react';

export const AgregarCat = ({onNewCategory}) => {

    const [first, setfirst] = useState('');

    const insertarTexto = (e) => {
        //setfirst(e)
        setfirst(e.target.value);
    }

    const alCapturarGif = (e) => {
        e.preventDefault();
        if (first.trim().length > 2){
            onNewCategory(first.trim());
        }
    }

    //d
  return (
    <div>
        <form onSubmit={alCapturarGif}>
            <input type="text" placeholder='Buscar Gifs' value={first} onChange={insertarTexto} />
            <h2>{first}</h2>
        </form>
    </div>
        
  )
}

