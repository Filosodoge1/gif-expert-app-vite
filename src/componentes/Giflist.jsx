import React from 'react'
import PropTypes from 'prop-types'

export const Giflist = ({title, url}) => {
  return (
    <div className='card'>
        <img src={url} alt={title}></img>
        <h1>{title}</h1>
    </div>
  )
}

Giflist.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}