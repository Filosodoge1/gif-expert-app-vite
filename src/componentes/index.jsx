export * from './AgregarCat';
export * from './GifGrid';
export * from './Giflist';