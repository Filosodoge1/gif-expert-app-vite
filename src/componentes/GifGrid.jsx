import React, { useEffect } from 'react'

import { useFetchGif } from '../hooks/useFetchGif';

import { Giflist } from './Giflist';

export const GifGrid = ({value}) => {
  
  const {first, isLoading} = useFetchGif(value);
  
  useEffect(() => {
    console.log(isLoading);
  }, [])
  
  return (
    <>
        
        <h1>{value}</h1>
        
        {
          
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className='card-grid'>
          {
            first.map( (img) => (<Giflist key={img.id} {...img} />) )
          }
        </div>
    </>
  )
}
