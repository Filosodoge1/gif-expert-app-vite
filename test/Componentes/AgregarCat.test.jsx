import { fireEvent, render, screen } from "@testing-library/react";
import { AgregarCat } from "../../src/componentes/AgregarCat";

describe('Probando el componente Agregar Categoria', () => {
    test('Debe de cambiar el valor de la caja de texto', () => {
        
        const {pag} = render(<AgregarCat onNewCategory={ () => {}} />);
        const cajatext = screen.getByRole('textbox');
        
        fireEvent.input( cajatext, {target: {value: 'Goku'}} );
        expect( cajatext.value ).toBe('Goku');

        
        expect(pag).toMatchSnapshot();
    });
});