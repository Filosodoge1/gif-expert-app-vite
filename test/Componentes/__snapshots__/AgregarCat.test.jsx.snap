// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente Agregar Categoria Debe de cambiar el valor de la caja de texto 1`] = `undefined`;
